service/collector/vault.go

```diff
@@ -0,0 +1,82 @@
+package collector
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	vault "github.com/hashicorp/vault/api"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// ExpireTimeKey is the data key provided by the secret when looking up the
+	// used Vault token. This key is specific to Vault as they define it.
+	ExpireTimeKey = "expire_time"
+	// ExpireTimeLayout is the layout used for time parsing when inspecting the
+	// expiration date of the used Vault token. This layout is specific to Vault
+	// as they define it.
+	ExpireTimeLayout = "2006-01-02T15:04:05"
+)
+
+var (
+	tokenExpireTimeDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("cert_operator", "vault", "token_expire_time"),
```

Comment:
The whole collector is basically the same as in `cluster-service`, but this metric name difference.

Reply:
Note that I need to add an alerting rule in `g8s-prometheus` as well still.